flang/lib/Lower/OpenMP.cpp

```diff
@@ -21,6 +23,32 @@
 
 #define TODO() llvm_unreachable("not yet implemented")
 
+static const Fortran::parser::Name *
+getDesignatorNameIfDataRef(const Fortran::parser::Designator &designator) {
+  const auto *dataRef{std::get_if<Fortran::parser::DataRef>(&designator.u)};
+  return dataRef ? std::get_if<Fortran::parser::Name>(&dataRef->u) : nullptr;
+}
+
+static void genObjectList(const Fortran::parser::OmpObjectList &objectList,
```

Comment:
Yes, I'm observing and marking these. Some of the Directive handling code can be shared(Maybe `LowerDirectives.cpp` in `lib/Lower`  dir ? or so..). But these are not on top of my priority list as of now.
Since they may break things(for short term) in Lowering of `OpenACC` or `OpenMP` of which both are developing rapidly and to some sort orthogonal to each other.